### DISCUSSION TITLE
company-etags backends not working with irony-mode

### DISCUSSION THREAD

**Discussion by gameoflord012** [company-mode/company-mode#1486] [Q&A] [answered]
Here is my basic etags and irony backends setup

```emacs-lisp
(use-package company
    :ensure t
    :config
    (setq company-idle-delay 0)
    (setq company-minium-prefix-length 2)
    (add-hook 'after-init-hook 'global-company-mode))

  (add-to-list 'company-backends 'company-etags) // adding company-etags before make it less priority than irony

  (use-package company-irony
    :ensure t
    :config
    (require 'company)
    (add-to-list 'company-backends 'company-irony))

  ;(add-to-list 'company-backends 'company-etags) // adding after make it work fine

  (use-package irony
    :ensure t
    :config
    (add-hook 'c++-mode-hook 'irony-mode)
    (add-hook 'c-mode-hook 'irony-mode)
    (add-hook 'objc-mode-hook 'irony-mode)
    (add-hook 'irony-mode-hook 'irony-cdb-autosetup-compile-options))
```

**Problem**
The etags backend doesn't give code completion when _company-etags_ is after _company-irony_.
I believe something get overriden or I might setup improperly.

**Comment by dgutov** [comment, marked as answer]
Hi! You might want to look at these questions and answers:

https://github.com/company-mode/company-mode/discussions/1249
https://github.com/company-mode/company-mode/discussions/1186

I'm also not sure if Irony does actually need augmentation with etags - it seems like a feature rich enough completion source that should include all the indexed methods. But you might have a special situation, I guess.

**Reply by gameoflord012** [reply]
Well, I want code completion with etags (for the code I've written) and also irony-mode (keyword suggestion).
Do you have better ideas?

**Reply by dgutov** [reply]
> and also irony-mode (keyword suggestion)

Irony-mode should provide full-blown completions, not just keywords. If it doesn't, that might be a configuration issue.

Anyway, see the two references above for how to mix backends.
